tests/unit/test_minion.py

```diff
@@ -318,7 +318,7 @@ def test_when_passed_start_event_grains(self):
         try:
             minion.tok = MagicMock()
             minion._send_req_sync = MagicMock()
-            minion._fire_master('Minion has started', 'minion_start')
+            minion._fire_master('Minion has started', 'minion_start', include_startup_grains=True)
```

Comment:
any chance we can also add a test that mirrors the test case found in issue https://github.com/saltstack/salt/issues/55870

with `start_event_grains` set, fire a non-startup event and ensure grains data is not in the event?

also heads up you have a lint failure

Reply:
Yep, +1 for the test to check that other events do not contain any grains.

Reply:
@Ch3LL thank you for having a look. I have added a test for this case as well.